### PR TITLE
RELATED: RAIL-3492 - Move more code from KD

### PR DESCRIPTION
**Pull request by lupko**
---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4258/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5868/ : SUCCESS in 15m 32s

File: libs/sdk-ui-dashboard/src/model/_staging/dashboard/dashboardLayout.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+function extractContentFromWidget(
+    widget: DashboardWidget,
+    insightsById: Record<string, IInsight>,
+): { type: WidgetType; content?: MeasurableWidgetContent } {
+    if (isInsightWidget(widget)) {
+        const insightRef = widget.insight;
+        const insight = insightsById[serializeObjRef(insightRef)];
+
+        if (!insight) {
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/model/_staging/dashboard/dashboardLayout.ts:26]: Detail: the if is redundant. Just
```
return {
    type: "insight",
    content: insight,
};
```
should be enough, right?

Comment (lupko) [reply libs/sdk-ui-dashboard/src/model/_staging/dashboard/dashboardLayout.ts]: fixed

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result
 - No issue found. Congratulations!

More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1486

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1222/ : FAILURE in 2m 38s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1498/ : SUCCESS in 16s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/252/ : SUCCESS in 12m 01s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5871/ : SUCCESS in 17m 32s

**Pull request merged**